controllers/platformnetwork_controller.go

```diff
@@ -661,38 +741,261 @@ func (r *PlatformNetworkReconciler) statusUpdateRequired(instance *starlingxv1.P
+	inSync := err == nil
+	if inSync != instance.Status.InSync || inSync != instance.Status.Reconciled {
+		r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "synchronization has changed to: %t", inSync)
 		instance.Status.InSync = inSync
+		instance.Status.Reconciled = inSync
+		err2 := r.Client.Status().Update(context.TODO(), instance)
+		if err2 != nil {
+			logPlatformNetwork.Error(err2, "failed to update platform network status")
+			return inSync, err2
+		}
+	}
+	return inSync, nil
+}
 
-		if !instance.Status.Reconciled {
-			// Record the fact that we have reached inSync at least once.
-			instance.Status.Reconciled = instance.Status.InSync
+func (r *PlatformNetworkReconciler) TriggerHostNetworkReconciliation(inSync bool, nwk_type string, host *hosts.Host, strategy string) bool {
+	if inSync {
+		logPlatformNetwork.Info(fmt.Sprintf("PlatformNetwork reconciliation successful for %s. Triggering host reconciliation.", nwk_type))
+		host_strategy := cloudManager.HostStrategyInfo{
+			StrategyRequired: strategy,
+			Host:             *host,
 		}
+		_ = r.CloudManager.SendHostReconciliationTrigger(host_strategy)
+		return true
+	}
+	return false
+}
 
-		if instance.Status.Reconciled {
-			instance.Status.ConfigurationUpdated = false
-			instance.Status.StrategyRequired = cloudManager.StrategyNotRequired
-			if instance.Status.DeploymentScope == cloudManager.ScopePrincipal {
-				r.CloudManager.SetResourceInfo(cloudManager.ResourcePlatformnetwork, "", instance.Name, instance.Status.Reconciled, cloudManager.StrategyNotRequired)
+// This method will try to reconcile both address pool as well as the network
+// and update the instance status accordingly.
+func (r *PlatformNetworkReconciler) ReconcileAddressPoolAndNetwork(client *gophercloud.ServiceClient, instance *starlingxv1.PlatformNetwork) (inSync bool, err error) {
+	err = r.ReconcileAddressPool(client, instance)
+	if err == nil {
+		if networkResourceRequired(instance) {
+			err = r.ReconcileNetwork(client, instance)
+		}
+	}
+
+	inSync, err2 := r.UpdateInsyncStatus(client, instance, err)
+	if err2 != nil {
+		return inSync, err2
+	}
+
+	return inSync, err
+}
+
+// ReconcileResource interacts with the system API in order to reconcile the
+// state of a data network with the state stored in the k8s database.
+func (r *PlatformNetworkReconciler) ReconcileResource(client *gophercloud.ServiceClient, instance *starlingxv1.PlatformNetwork) (err error) {
+	oldStatus := instance.Status.DeepCopy()
+
+	if instance.DeletionTimestamp.IsZero() {
+		update_pn, err := r.PlatformNetworkUpdateRequired(client, instance)
+		if err != nil {
+			return err
+		}
+		if instance.Status.DeploymentScope == cloudManager.ScopeBootstrap {
+			inSync := !update_pn
+			if instance.Spec.Type == cloudManager.MgmtNetworkType || instance.Spec.Type == cloudManager.OAMNetworkType || instance.Spec.Type == cloudManager.AdminNetworkType {
```

Comment:
If a line feels too long, it should be refactored instead of broken.

Reply:
> If a line feels too long, it should be refactored instead of broken.

Done.